Composer/packages/client/src/components/Notifications/useSurveyNotification.ts

```diff
@@ -31,10 +31,12 @@ const buildUrl = (info: MachineInfo) => {
 };
 
 export const useSurveyNotification = () => {
-  const { addNotification, deleteNotification } = useRecoilValue(dispatcherState);
+  const { addNotification, deleteNotification, setSurveyEligibility } = useRecoilValue(dispatcherState);
   const surveyEligible = useRecoilValue(surveyEligibilityState);
   const machineInfo = useRecoilValue(machineInfoState);
 
+  setSurveyEligibility();
```

Comment:
This would keep calling the dispatcher quite a few times @beyackle . We should wrap it in a useEffect. Is there a reason you are placing it here?

Reply:
This needs to run every time we could potentially display the notification. This dispatcher runs the logic to determine whether the user is eligible for a survey or not, including setting that to `false` once the survey has been taken. If we didn't do that, then the notification would just keep showing up because the eligibility would stay true.

Reply:
Fixed - found a better place to run this logic.